packages/@lwc/engine-core/src/framework/template.ts

```diff
@@ -264,6 +277,7 @@ function buildParseFragmentFn(
                 hasScopedStyles && hasStyleToken ? ` class="${stylesheetTokenToRender}"` : '';
             const attrToken =
                 hasStyleToken && isSyntheticShadow ? ' ' + stylesheetTokenToRender : '';
+            const exprClassToken = process.env.IS_BROWSER ? classAttrToken : classToken;
```

Comment:
A comment here explaining the difference between `classAttrToken` and `classToken` and why one is used in the browser would be heplful.

Reply:
Added a [comment](https://github.com/salesforce/lwc/pull/4071/commits/c3dddb4565966b351413c05bf1596a8d4c99a953#diff-9d36a520650dfa1dd13874ab3870b6f688303b9e5860066178098aadcf45a860R280-R283)!